### PR TITLE
internal/cli: colorize flags in help text and nested subcommands

### PR DESCRIPTION
**Pull request by mitchellh**
This modifies our CLI help output to do two things:

  1. We always highlighted commands such as "waypoint foo", but nested
     subcommands "waypoint foo bar" weren't highlighted. This fixes
     that.

  2. Flags in the help text body such as -foo and -foo-bar are now
     highlighted as well. This is only in the help text summary and not
     the flag documentation itself.

Screenshots (sorry my colors are very muted so its hard to see):

Flag highlight (see -waypoint-hcl):

CleanShot 2021-01-19 at 12 19 45@2x

Nested subcommand highlight:

CleanShot 2021-01-19 at 12 19 57@2x

### CODE REVIEW

**Review by briancain — APPROVED** [review]
:art:

File: internal/cli/help.go [hunk 1]
```diff
@@ -128,7 +165,8 @@ func (c *helpCommand) HelpTemplate() string {
 
 var (
 	reHelpHeader = regexp.MustCompile(`^[a-zA-Z0-9_-].*:$`)
-	reCommand    = regexp.MustCompile(`"waypoint \w+"`)
+	reCommand    = regexp.MustCompile(`"waypoint (\w\s?)+"`)
+	reFlag       = regexp.MustCompile(`(\s|^)(-[\w-]+)(\s|$)`)
```
Comment (krantzinator) [inline internal/cli/help.go:169]: I needed the below to get the `"-accept-tos"` bit in `install -h` to highlight.
```suggestion
	reFlag       = regexp.MustCompile(`(\s|^|")(-[\w-]+)(\s|$|")`)
```

Comment (krantzinator) [reply internal/cli/help.go:169]: We could instead remove all `"` around flags in help texts.

Comment (krantzinator) [reply internal/cli/help.go:169]: Or just `/W` and assume we won't ever have something like `$-this-is-not-a-flag#`

Comment (mitchellh) [reply internal/cli/help.go]: Ah yes. No this is fine with me!

**Pull request merged**